ocfweb/docs/docs/staff/startertasks.md

```diff
@@ -43,48 +62,174 @@ you know how to use it:
-which repository something is in, you can use [OCF
-   Sourcegraph](https://sourcegraph.ocf.berkeley.edu) to search across all repositories!
+   which repository something is in, you can use [OCF Sourcegraph][sourcegraph]
+   to search across all repositories!
 
 ## Play with your webspace
-Every OCF account has [web
-hosting](https://www.ocf.berkeley.edu/docs/services/web/) enabled at
-https://www.ocf.berkeley.edu/~yourusername. As an example, check out
-[ckuehl’s website](https://www.ocf.berkeley.edu/~ckuehl/).
+Every OCF account has [[web hosting|doc services/web]] enabled at
+https://www.ocf.berkeley.edu/~yourusername. As an example, check out [ckuehl’s
+website](https://www.ocf.berkeley.edu/~ckuehl/).
 
 1. Add some files to your webspace and preview it in your web browser.
 2. (optional) Most student groups that host with the OCF use WordPress. Install
    WordPress in your webspace. (hint: instructions for this are on our website)
 
 ## Play with ocflib\*
-[ocflib](https://github.com/ocf/ocflib) is a Python library we maintain which is
-installed on every OCF host. For this exercise, you won’t need to make
-modifications to ocflib.
-
-1. Log into supernova and start an IPython3 shell (the `ipython3` command).
-2. Run `import ocflib`.
-3. Use ocflib functions to get the following information in your interactive
-   Python shell:
-
-    a. Get the list of signatories for the OCF, and your favorite student org.
-    (hint: look at the signat source code from before!)
-
-    b. Find the toner levels of each printer.
+[ocflib][ocflib] is a Python library we maintain which is installed on every OCF
+host. For this exercise, you won’t need to make modifications to ocflib.
+
+1. [[Log into supernova|doc staff/procedures/ssh-supernova]] and start an
+   IPython3 shell (the `ipython3` command).
+2. To make sure things are working:
+   1. Run `import ocflib.lab.staff_hours` to import utilities relating to
+      [[staff hours|staff-hours]].
+   2. Run `staffhours = ocflib.lab.staff_hours.get_staff_hours()` to get the
+      list of staff hours.
+   3. Take a look at the list `staffhours` and check that it matches the info
+      on [[the staff hours page|staff-hours]]. For instance, you should be able
+      to run `staffhours[0]` to see the info corresponding to the first staff
+      hour entry on the page.
+3. Your turn! Use ocflib functions to get the following information in your
+   interactive Python shell:
+   1. Get the list of signatories for the OCF, and your favorite student org.
+   (hint: look at the signat source code from before!)
+   2. Find the toner levels of each printer.
+   3. Find the hours the OCF was open on your most recent birthday.
+
+   If you get stuck, don't forget that you can search the [GitHub repo][ocflib]
+   or [Sourcegraph][sourcegraph] to find more info.
+
+[ocflib]: https://github.com/ocf/ocflib
+[sourcegraph]: https://sourcegraph.ocf.berkeley.edu
+
+## Make a pull request!\*
```

Comment:
I like this idea for an exercise, but it's way too hand-holdy. As I mentioned before, the point of this page is for people to learn this stuff on their own and at their own pace. It's fine to give minimal direction here-- maybe mention that you'll need a GitHub account, you'll have to fork the repo and make a pull request, and you should test your changes before submitting a PR-- but other than that the reader should be able to find out how to do this their own. There are plenty of guides for this workflow on the internet.

We do have OCF-isms that you can't find on the internet, like needing to install pre-commit hooks and running `make dev` to test on supernova.. But this information is in the ocfweb README (which we could provide as a hint). If people forget it, we can remind them in the PR comments.

Reply:
Part of my concern is that **making a PR from a fork is a complicated process**, and to be honest there are _not_ good guides on how to do it. I don't think it's really reasonable to expect that people need to have a full understanding of git that includes handling multiple remotes and branches in order to make a contribution. That's the primary reason this guide is so detailed. I would like to strike a better balance but I'm not really sure how to do that.

git kind of just sucks for new people and I'm not sure how to reduce that pain. If you google "how to make a pull request" you'll get a series of results that are somehow worse than this. (Either more hand-holdy while being less descriptive or assuming a lot of knowledge of git already.)

Anyway, I'll try to take another pass at this and make it a bit more self-directed. I can probably remove nearly all the commands and just say "do this" with a link to GitHub docs.

Reply:
It is safer to err on the side of being prescriptive about the steps, especially basic ones. Git is a major hurdle for most new contributors and is just one of the tools that is part of the 'OCF contribution workflow'. They shouldn't be spending too much time getting set up (causing them to drop out).